test/e2e/storage/testsuites/fsgroupchangepolicy.go

```diff
@@ -0,0 +1,278 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package testsuites
+
+import (
+	"fmt"
+	"strconv"
+
+	"github.com/onsi/ginkgo"
+	v1 "k8s.io/api/core/v1"
+	errors "k8s.io/apimachinery/pkg/util/errors"
+	"k8s.io/kubernetes/test/e2e/framework"
+	e2epod "k8s.io/kubernetes/test/e2e/framework/pod"
+	e2eskipper "k8s.io/kubernetes/test/e2e/framework/skipper"
+	e2evolume "k8s.io/kubernetes/test/e2e/framework/volume"
+	"k8s.io/kubernetes/test/e2e/storage/testpatterns"
+	storageutils "k8s.io/kubernetes/test/e2e/storage/utils"
+	utilpointer "k8s.io/utils/pointer"
+)
+
+type fsGroupChangePolicyTestSuite struct {
+	tsInfo TestSuiteInfo
+}
+
+var _ TestSuite = &fsGroupChangePolicyTestSuite{}
+
+// InitFsGroupChangePolicyTestSuite returns fsGroupChangePolicyTestSuite that implements TestSuite interface
+func InitFsGroupChangePolicyTestSuite() TestSuite {
+	return &fsGroupChangePolicyTestSuite{
+		tsInfo: TestSuiteInfo{
+			Name: "fsgroupchangepolicy",
+			TestPatterns: []testpatterns.TestPattern{
+				testpatterns.DefaultFsDynamicPV,
+			},
+			SupportedSizeRange: e2evolume.SizeRange{
+				Min: "1Mi",
+			},
+		},
+	}
+}
+
+func (s *fsGroupChangePolicyTestSuite) GetTestSuiteInfo() TestSuiteInfo {
+	return s.tsInfo
+}
+
+func (s *fsGroupChangePolicyTestSuite) SkipRedundantSuite(driver TestDriver, pattern testpatterns.TestPattern) {
+	skipVolTypePatterns(pattern, driver, testpatterns.NewVolTypeMap(testpatterns.CSIInlineVolume, testpatterns.GenericEphemeralVolume))
+}
+
+func (s *fsGroupChangePolicyTestSuite) DefineTests(driver TestDriver, pattern testpatterns.TestPattern) {
+	type local struct {
+		config        *PerTestConfig
+		driverCleanup func()
+		driver        TestDriver
+		resource      *VolumeResource
+	}
+	var l local
+	ginkgo.BeforeEach(func() {
+		dInfo := driver.GetDriverInfo()
+		if !dInfo.Capabilities[CapFsGroup] {
+			e2eskipper.Skipf("Driver %q does not support FsGroup - skipping", dInfo.Name)
+		}
+
+		if pattern.VolMode == v1.PersistentVolumeBlock {
+			e2eskipper.Skipf("Test does not support non-filesystem volume mode - skipping")
+		}
+
+		if pattern.VolType != testpatterns.DynamicPV {
+			e2eskipper.Skipf("Suite %q does not support %v", s.tsInfo.Name, pattern.VolType)
+		}
+
+		_, ok := driver.(DynamicPVTestDriver)
+		if !ok {
+			e2eskipper.Skipf("Driver %s doesn't support %v -- skipping", dInfo.Name, pattern.VolType)
+		}
+
+		if framework.NodeOSDistroIs("windows") {
```

Comment:
I think this check doesn't work because the framework is not initialized yet. Maybe we need to add "[LinuxOnly]" to the testsuite name instead

Reply:
thanks for catching this. I have moved the skip check to init() which is called inside an It()